core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -274,9 +275,16 @@ object ExternalStoreUtils extends Logging {
         false
       case _ => false
     }
-    val allPoolProps = getAllPoolProperties(url, driver,
-      poolProps, hikariCP, isEmbedded)
-    ConnectionProperties(url, driver, dialect, allPoolProps,
+    val allPoolProps = getAllPoolProperties(url, driver, poolProps, hikariCP, isEmbedded)
+    val EMPTY_URL = ""
+    val urlSecureSuffix = if (session.isDefined) {
+      val user = session.get.conf.getOption(Attribute.USERNAME_ATTR)
+      val password = session.get.conf.getOption(Attribute.PASSWORD_ATTR)
+      if (user.isDefined && password.isDefined) {
+        s";user=${user.get};password=${password.get};"
+      } else EMPTY_URL
+    } else EMPTY_URL
+    ConnectionProperties(url, urlSecureSuffix, driver, dialect, allPoolProps,
```

Comment:
Again I think filling into connProps and executorConnProps will be better and avoid the changes related to addition of "urlSecureSuffix".